Update `inv shell` for Django

sloria opened the pull request:
<!-- Before submit your Pull Request, make sure you picked
     the right branch:

     - For hotfixes, select "master" as the target branch
     - For new features, select "develop" as the target branch
     - For release feature fixes, select the relevant release branch (release/X.Y.Z) as the target branch -->

## Purpose

Updates and improves the `inv shell` (which is equivalent to `python manage.py osf_shell`.

## Changes

<!-- Briefly describe or list your changes  -->

## Side effects

This depends on a fork of `django-extensions` that has a modification to allow adding additional imports at runtime (https://github.com/django-extensions/django-extensions/pull/1039). This change has been merged upstream, so we can remove the fork when django-extensions cuts another release.

Devs will need to update their requirements: `docker-compose up requirements` and/or `pip install -r requirements.txt`

## Ticket

<!-- Link to JIRA ticket, if applicable e.g. https://openscience.atlassian.net/browse/OSF-1234 -->

## Notes

- I'm sending this as a hotfix so that the behavior of `inv shell` is consistent on master and develop. It's a low-risk PR, anyway.
- I had issues with garbage characters showing up when entering characters into IPython. Seems to be an issue with invoke<=0.13.0. Upgrading invoke locally (0.15.0) fixed the issue. I'll send another PR to develop upgrading our pinned version.

icereval commented on tasks/__init__.py:
```diff
@@ -109,152 +108,12 @@ def adminserver(ctx, port=8001, host='127.0.0.1', pty=True):
     ctx.run(cmd, echo=True, pty=pty)
 
-
-SHELL_BANNER = """
```
:+1: keep banner and put in shell/shell_plus

icereval commented on tasks/__init__.py:
```diff
@@ -109,152 +108,12 @@ def adminserver(ctx, port=8001, host='127.0.0.1', pty=True):
-    return
-
+def shell(ctx, quiet=False):
+    cmd = 'DJANGO_SETTINGS_MODULE="api.base.settings" python manage.py shell_plus'
```
:+1: add transaction opt-out to shell/shell_plus

sloria commented:
Added back the automatic transaction behavior, added back the banner, and better formatting of the imports. Also added a `OSF_SHELL_USER_IMPORTS` setting so that devs can add more imports.

![2__tmux](https://cloud.githubusercontent.com/assets/2379650/24719821/196925ba-1a09-11e7-8bb4-652c72c1b4ee.png)
